completions/patch

```diff
@@ -50,7 +50,11 @@ _comp_cmd_patch()
     [[ $was_split ]] && return
 
     if [[ $cur == -* ]]; then
-        _comp_compgen_help
+        if [[ $OSTYPE == *@(darwin|bsd)* ]]; then
+            _comp_compgen_usage
+        else
+            _comp_compgen_help
+        fi
```

Comment:
To me, the preferred pattern for cases like these is
```suggestion
        _comp_compgen_help || _comp_compgen_usage
```

This frees us from hardwiring "known" OS dependencies, as well as helps with cases for which detection isn't that feasible, such as Alpine systems where a command might be a GNU one supporting long options (_comp_compgen_help applies), or a busybox one that supports only short ones (_comp_compgen_usage applies).